src/drivers/bitbucket_cloud.js

```diff
@@ -0,0 +1,77 @@
+const fetch = require('node-fetch');
+const FormData = require('form-data');
+const { URL, URLSearchParams } = require('url');
+
+const { fetch_upload_data } = require('../utils');
+
+class BitBucketCloud {
+  constructor(opts = {}) {
+    const { repo, token } = opts;
+
+    if (!token) throw new Error('token not found');
+    if (!repo) throw new Error('repo not found');
+
+    this.token = token;
+    this.repo = repo;
+
+    const { protocol, host, pathname } = new URL(this.repo);
+    this.repo_origin = `${protocol}//${host}`;
+    this.api = 'https://api.bitbucket.org/2.0';
+    this.project_path = encodeURIComponent(pathname.substring(1));
+  }
+
+  async comment_create(opts = {}) {
+    const { project_path } = this;
+    const { commit_sha, report} = opts;
+
+    // Print some variables
+    const endpoint = `/repositories/${project_path}/commit/${commit_sha}/comments/`;
+    const body= JSON.stringify({"content": {"raw": report}});
+
+    const output = await this.request({ endpoint, method: 'POST', body });
+
+    return output;
+  }
+
+  async check_create() {
+    throw new Error('BitBucket Cloud does not support check!');
+  }
+
+  async upload(opts = {}) {
```

Comment:
Does BB support any kind of upload?

Reply:
there are artifacts, but they're impermanent- 14 days https://support.atlassian.com/bitbucket-cloud/docs/use-artifacts-in-steps/
Makes me think there are very few advantages to integrating with CML.